specs/governance/governor.md

```diff
@@ -0,0 +1,730 @@
+    - [`propose`](#propose)
+    - [`proposeWithModule`](#proposewithmodule)
+    - [`relay`](#relay)
+    - [`setModuleApproval`](#setmoduleapproval)
+    - [`setProposalDeadline`](#setproposaldeadline)
+    - [`setProposalThreshold`](#setproposalthreshold)
+    - [`setVotingDelay`](#setvotingdelay)
+    - [`setVotingPeriod`](#setvotingperiod)
+    - [`updateQuorumNumerator`](#updatequorumnumerator)
+  - [Getters](#getters)
+    - [`approvedModules`](#approvedmodules)
+    - [`getProposalType`](#getproposaltype)
+    - [`getVotes`](#getvotes)
+    - [`getVotesWithParams`](#getvoteswithparams)
+    - [`hasVoted`](#hasvoted)
+    - [`hashProposal`](#hashproposal)
+    - [`hashProposalWithModule`](#hashproposalwithmodule)
+    - [`manager`](#manager)
+    - [`name`](#name)
+    - [`proposalDeadline`](#proposaldeadline)
+    - [`proposalSnapshot`](#proposalsnapshot)
+    - [`proposalThreshold`](#proposalthreshold)
+    - [`proposalVotes`](#proposalvotes)
+    - [`quorum`](#quorum)
+    - [`quorumDenominator`](#quorumdenominator)
+    - [`quorumNumerator`](#quorumnumerator)
+    - [`quorumNumerator` (current)](#quorumnumerator-current)
+    - [`state`](#state)
+    - [`supportsInterface`](#supportsinterface)
+    - [`token`](#token)
+    - [`version`](#version)
+    - [`votableSupply`](#votablesupply)
+    - [`votableSupply` (specific block)](#votablesupply-specific-block)
+    - [`votingDelay`](#votingdelay)
+    - [`votingPeriod`](#votingperiod)
+    - [`weightCast`](#weightcast)
+  - [Events](#events)
+    - [`ProposalCanceled`](#proposalcanceled)
+    - [`VoteCast`](#votecast)
+    - [`VoteCastWithParams`](#votecastwithparams)
+    - [`ProposalTypeUpdated`](#proposaltypeupdated)
+    - [`ProposalExecuted`](#proposalexecuted)
+    - [`ProposalCreated`](#proposalcreated)
+    - [`ProposalDeadlineUpdated`](#proposaldeadlineupdated)
+    - [`ProposalThresholdSet`](#proposalthresholdset)
+    - [`VotingDelaySet`](#votingdelayset)
+    - [`VotingPeriodSet`](#votingperiodset)
+    - [`QuorumNumeratorUpdated`](#quorumnumeratorupdated)
+- [Proposal Types](#proposal-types)
+  - [Interface](#interface-1)
+- [Modules](#modules)
+  - [Interface](#interface-2)
+- [Security Considerations](#security-considerations)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The `Governor` contract implements the core governance logic for creating, voting, and executing proposals.
+This contract uses the `GovernanceToken` contract for voting power snapshots, and the `ProposalTypesConfigurator`
+for proposal types.
+
+## Constants
+
+| Constant          | Value                           | Description |
+| ----------------- | ------------------------------- | ----------- |
+| `COUNTING_MODE` | `support=bravo&quorum=against,for,abstain&params=modules` | The configuration for supported values for `castVote` and how votes are counted, which is also consumed by UIs to show correct vote option and interpret results. |
```

Comment:
What does bravo mean here?

Reply:
refers to the style of governor bravo voting options (against, for, and abstain). clarified this in the `castVote` function: https://github.com/ethereum-optimism/specs/pull/315/commits/09c3cc44457da224e94ae8f3576d4f946432fefc